pkg/agent/nodeportlocal/k8s/npl_controller.go

```diff
@@ -422,9 +422,14 @@ func (c *NPLController) handleAddUpdatePod(key string, obj interface{}) error {
 			podPorts[port] = struct{}{}
 			portData := c.portTable.GetEntryByPodIPPort(podIP, int(cport.ContainerPort))
 			if portData == nil { // rule does not exist
-				nodePort, err = c.portTable.AddRule(podIP, port)
-				if err != nil {
-					return fmt.Errorf("failed to add rule for Pod %s: %v", key, err)
+				if int(cport.HostPort) > 0 {
+					klog.V(2).Infof("Host Port is defined for Pod, thus extra NPL port is not allocated: %s", key)
```

Comment:
maybe a bit more context is needed in the log message (e.g. the container port), since a Pod can have multiple containers, with multiple ports, and hostPort is specific to a given port?

Reply:
Added container name